src/main/webapp/app/service/firebase/firebase-gene-review-service.ts

```diff
@@ -140,7 +149,28 @@ export class FirebaseGeneReviewService {
     try {
       await this.firebaseRepository.update('/', updateObject);
     } catch (error) {
-      throw new SentryError('Failed to accept changes in review mode', { hugoSymbol, reviewLevels, isGermline, isAcceptAll, updateObject });
+      throw new SentryError('Failed to accept changes in review mode', {
+        hugoSymbol,
+        reviewLevels,
+        isGermline,
+        isAcceptAll,
+        updateObject,
+      });
+    }
+
+    // We are deleting last because the indices will change after deleting from array.
+    try {
+      // Todo: We should use multi-location updates for deletions once all our arrays use firebase auto-generated keys
+      // instead of using sequential number indices.
+      /* eslint-disable no-console */
+      console.log(itemsToDelete);
+      for (const pathType of [FIREBASE_PATH_TYPE.TREATMENT_LIST, FIREBASE_PATH_TYPE.TUMOR_LIST, FIREBASE_PATH_TYPE.MUTATION_LIST]) {
+        for (const [firebasePath, deleteIndices] of Object.entries(itemsToDelete[pathType])) {
+          await this.firebaseRepository.deleteFromArray(firebasePath, deleteIndices);
+        }
+      }
```

Comment:
I see here we are trying to delete deeper list first to avoid mismatch. But how are you addressing deleting/updating the next index?

Reply:
Good question. 
`firebaseRepository.deleteFromArray(firebasePath, deleteIndices)` will perform the deletion of the array of index in a transaction. The index will never be stale because we loop through the original list to construct a new firebase list. 

Check out the code: https://github.com/oncokb/oncokb-transcript/blob/rc/src/main/webapp/app/stores/firebase/firebase-repository.ts#L83-L92